src/utils.jl

```diff
@@ -69,11 +69,10 @@ function invwsumsq(w::AbstractVector, a::AbstractVector)
 end
 
 function colwise_dot!(r::AbstractArray, a::AbstractMatrix, b::AbstractMatrix)
-    n = length(r)
-    @check_argdims n == size(a, 2) == size(b, 2) && size(a, 1) == size(b, 1)
-    for j = 1:n
+    @check_argdims(axes(a) == axes(b))
+    for j in axes(a, 2)
         v = zero(promote_type(eltype(a), eltype(b)))
-        @simd for i = 1:size(a, 1)
+        @simd for i in axes(a, 1)
             @inbounds v += a[i, j]*b[i, j]
         end
         r[j] = v
```

Comment:
Similar to my comments in the other PR, `j` is not necessarily a valid index for `r`. We actually want something like
```suggestion
    n = length(r)
    @check_argdims n == size(a, 2) == size(b, 2) && size(a, 1) == size(b, 1)
    for (i, ai, bi) in zip(eachindex(r), eachol(a), eachcol(b))
        r[i] = dot(ai, bi)
```

Reply:
Per my comment on other PR we know it is because we did `check_argdims(axes(a) == axes(b))`

Reply:
The check `check_argdims(axes(a) == axes(b))` only helps with ensuring that `i, j` are valid indices for `a` and `b`. But it seems it doesn't guarantee that `j` can be used as index for `r`.